src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp

```diff
@@ -233,9 +225,16 @@ void YARPRobotStatePublisherModule::onRead(yarp::rosmsg::sensor_msgs::JointState
     iDynTree::JointIndex jntIndex;
     for (size_t i=0; i < v.name.size(); i++)
     {
-        jntIndex = model.getJointIndex(v.name[i]);
-
-        if (!(model.getJoint(jntIndex)->getNrOfDOFs()) || jntIndex == iDynTree::JOINT_INVALID_INDEX)
+        for (size_t j=0; j < model.getNrOfDOFs(); j++)
+        {
+            if (v.name[i] == model.getJointName(j))
+            {
+                jntIndex = model.getJointIndex(v.name[i]);
+                break;
+            }
+        }
+
```

Comment:
Can't we just leave the `jntIndex = model.getJointIndex(v.name[i]);` line, and just check if `jntIndex` is `iDynTree::JOINT_INVALID_INDEX` ?
`

Reply:
Yes, force pushed a new commit that removes only the joint size check and doesn't modify joint index handling.